server/route.go

```diff
@@ -142,10 +201,10 @@ func (r *Route) Close() error {
 	return ErrInvalidRoute
 }
 
-func (r *Route) equals(other *Route) bool {
-	return r.Path == other.Path &&
-		r.UserID == other.UserID &&
-		r.ApplicationID == other.ApplicationID
+// Equal will check if the route path is matched and all the parameters or just a
+// subsent of specific parameters between the routes
```

Comment:
// Equal checks if the route's path and all the route params having the given keys are matching.